url_text_extractor.py

```diff
@@ -2,12 +2,32 @@
 from bs4 import BeautifulSoup
 
 
-def extract_text_from_url(url: str) -> str:
+def extract_images_from_html(soup: BeautifulSoup) -> list:
+    images = []
+    # Extract from <img> tags
+    for img in soup.find_all('img'):
+        if img.get('src'):
+            images.append(img.get('src'))
+    # Extract from any tag with a style attribute containing 'background-image'
+    for tag in soup.find_all(style=True):
+        style = tag['style']
+        if 'background-image' in style:
+            url_start = style.find('url(') + 4
+            url_end = style.find(')', url_start)
+            image_url = style[url_start:url_end].strip('"\'')
```

Comment:
Is this a good way to extract background images?

Reply:
The current method of extracting background images from the `style` attribute is functional but not robust. It directly parses the `style` string to find URLs, which can be error-prone if the `style` attribute contains more complex CSS or variations in formatting.

A more reliable approach would be to use regular expressions (regex) to accurately extract URLs from the `style` attributes. This method can handle variations in whitespace, the presence of additional CSS properties, and different quoting styles around the URL.

I will update the code to use regex for extracting background images and amend the PR accordingly.